common/src/Microsoft.Azure.IIoT.Hub.Module.Client/src/Default/MqttClient/ManagedMqttClientStorage.cs

```diff
@@ -0,0 +1,62 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.Hub.Module.Client.Default.MqttClient {
+    using MQTTnet.Extensions.ManagedClient;
+    using Newtonsoft.Json;
+    using Serilog;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    /// <inheritdoc />
+    public class ManagedMqttClientStorage : IManagedMqttClientStorage {
+        private readonly string _stateFile;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Create client storage.
+        /// </summary>
+        public ManagedMqttClientStorage(string stateFile, ILogger logger) {
+            if (string.IsNullOrWhiteSpace(stateFile)) {
+                throw new ArgumentException($"'{nameof(stateFile)}' cannot be null or whitespace.", nameof(stateFile));
+            }
+
+            _stateFile = stateFile;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <inheritdoc />
+        public async Task<IList<ManagedMqttApplicationMessage>> LoadQueuedMessagesAsync() {
+            if (File.Exists(_stateFile)) {
```

Comment:
probably we should add a log in case the _stateFile does not exist

Reply:
Added